api/migrations/20230207224139_mongoose-adding-funding-category.js

```diff
@@ -0,0 +1,126 @@
+import loggerFactory from '../logger/index.js';
+import { setup, teardown } from '../db/mongodb.js';
+import { APD, Budget } from '../models/index.js';
+import {
+  APD_TYPE,
+  FUNDING_CATEGORY_TYPE,
+  calculateBudget,
+  deepCopy
+} from '@cms-eapd/common';
+
+const logger = loggerFactory(
+  'mongoose-migrate/migrate-adding-funding-category'
+);
+
+const updateFundingCategory = ({ federal = null, state = null } = {}) => {
+  if (federal === 90 && state === 10) {
+    return FUNDING_CATEGORY_TYPE.ddi;
+  }
+  if (federal === 75 && state === 25) {
+    return FUNDING_CATEGORY_TYPE.mando;
+  }
+  return null;
+};
+
+/**
+ * Update APDs to have a fundingCategory field in costAllocation.
+ * HITECH APDs keep their existing ffp split and have a null fundingCategory.
+ * MMIS APDs have their ffp split reset and have a null fundingCategory.
+ */
+export const up = async () => {
+  // Grab all APDs
+  await setup();
+  const apds = await APD.find({
+    'activities.costAllocation.$[].ffp.fundingCategory': { $exists: false }
+  }).lean();
+  logger.info(`Updating ${apds.lengths} APDs`);
+
+  const apdIds = [];
+  const budgetIds = [];
+
+  const updatedApds = apds.map(apd => {
+    apdIds.push(apd._id);
+    if (apd.budget) {
+      budgetIds.push(apd.budget);
+    }
+    const apdJSON = deepCopy(apd);
+    const years = apd.years;
+    const apdType = apd.apdType;
+
+    apdJSON.keyStatePersonnel.keyPersonnel =
+      apdJSON.keyStatePersonnel.keyPersonnel.map(keyPersonnel => ({
+        ...keyPersonnel,
+        split: years.reduce(
+          /**
```

Comment:
Do you need to update the fed and state again? I would think only the fundingCategory would need to be updated. Either way, update this inline-doc to include the fundingCategory change

Reply:
I'll switch it to only update the funding category